jormungandr/src/jrpc/eth_types/filter.rs

```diff
@@ -0,0 +1,143 @@
+use super::{block_number::BlockNumber, log::Log};
+use chain_evm::ethereum_types::{H160, H256};
+use serde::{Deserialize, Serialize, Serializer};
+
+/// Variadic value
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum VariadicValue<T> {
+    /// Single
+    Single(T),
+    /// List
+    Multiple(Vec<T>),
+    /// None
+    Null,
+}
+
+/// Filter Address
+pub type FilterAddress = VariadicValue<H160>;
+/// Topic, supports `A` | `null` | `[A,B,C]` | `[A,[B,C]]` | [null,[B,C]] | [null,[null,C]]
```

Comment:
`VariadicValue<VariadicValue>>` allows for other formats as well (e.g. [A, B, C, D]), if that's well defined it's ok, otherwise please restrict valid values

Reply:
Yes, sure it is possible to have a `[A, B, C, D]` or `[A, B, C, D, F]`, we don't have a restrictions of the vec size

Reply:
`[A, B, C, D]` it is array of `H256` hashes

Reply:
I have followed by this description of the `Filter` returning value https://eth.wiki/en/json-rpc/API, `eth_newFilter` method